route_mappings.go

```diff
@@ -254,22 +240,54 @@ func (a *App) ANY(p string, h Handler) {
 	g.GET("/users/:user_id, APIUserShowHandler)
 */
 func (a *App) Group(groupPath string) *App {
+	// TODO: move this function to app.go or home.go eventually.
 	g := New(a.Options)
+	// keep them for v0 compatibility
 	g.Prefix = path.Join(a.Prefix, groupPath)
 	g.Name = g.Prefix
 
+	// for Home structure
+	g.prefix = path.Join(a.prefix, groupPath)
+	g.host = a.host
+	g.name = g.prefix
+
 	g.router = a.router
 	g.RouteNamer = a.RouteNamer
 	g.Middleware = a.Middleware.clone()
 	g.ErrorHandlers = a.ErrorHandlers
-	g.root = a
-	if a.root != nil {
-		g.root = a.root
-	}
+
+	g.app = a.app  // will replace g.root
+	g.root = g.app // will be deprecated
```

Comment:
since g.root is a private field it could already be deprecated, while the public fields of `App` should not be removed but only marked as deprecated.

Reply:
As far as I remember, there is no deprecated *public* field. As you said, the private field could be deprecated right now but just keep them since (1) they are used in some other places and (2) want to track the change for v1 in the code context.

This change was introduced to fix the breaking change introduced by #2185 as a quick fix (with the smallest changes) and keep buffalo's existing behavior the same as now, and I think more effect could be focused on v1 with unavoidable breaking structure change.